### PR TITLE
Building Docker Image and Pusing to ECR

### PR DESCRIPTION
**Pull request by shooshmand-sol**
### What is the purpose of this change?
Once we merge into main, we want to build a docker image and push to ECR 

### How was this change implemented?
- Calling service/application/docker/buildEventManagementAgentDocker.sh script to create the docker image
- Tagging the image with `main` and `main-<pom-file-version>-gitsha` and pushing to ECR.

### How was this change tested?
https://github.com/SolaceLabs/event-management-agent/actions/runs/5246575481

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
   Test_Build:
     name: Run Tests and Deploy
     runs-on: ubuntu-latest
+    environment: ${{ github.ref=='refs/heads/main' && 'build_main' || 'build_pr' }}
```
Comment (shooshmand-sol) [inline .github/workflows/build.yaml:21]: dynamically setting environment to protect secrets.

File: service/application/docker/buildEventManagementAgentDocker.sh [hunk 2]
```diff
@@ -105,13 +105,14 @@ export BASE_IMAGE=event-management-agent-base:${BASE_IMAGE_TAG}
 export GITHASH=$(git rev-parse HEAD)
 export GITBRANCH=$(git branch --show-current)
 export BUILD_TIMESTAMP=$(date -u)
-cp ../target/event-management-agent-1.0.0-SNAPSHOT.jar .
+export JAR_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout --file ../pom.xml)
```
Comment (shooshmand-sol) [inline service/application/docker/buildEventManagementAgentDocker.sh:108]: This maven command returns the veriosn in the pom file.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (shooshmand-sol) [reply .github/workflows/build.yaml:21]: for main branch use `build_main` environment, for PRs use `build_pr`

**Review by johnvincentcorpuz — APPROVED** [review]
Looks good. 👍

**Review by antonysavio-sol — APPROVED** [review]
👍

**Pull request merged**